tests/foreman/api/test_activationkey.py

```diff
@@ -424,7 +424,7 @@ def test_positive_remove_user():
     """
     password = gen_string('alpha')
     user = entities.User(password=password, login=gen_string('alpha'), admin=True).create()
-    cfg = get_nailgun_config()
+    cfg = deepcopy(default_sat.nailgun_cfg)
```

Comment:
that nailgun_cfg object doesn't exist until the Satellite's API property is evaluated.

Reply:
yeah that's what I was worried about. Its basically chance that its been called by the time this is done.

Thoughts on the pattern overall? Leave this commit out of the PR for further discussion? Add a call to `default_sat.api` before referencing the attribute in all these uses? Add a call to the `Satellite.__init__` to `self.api`?

Reply:
Discussed strategies a bit offline, with @JacobCallahan having some good ideas for where to init this attribute effectively.

I think I'm going to pull the commit for this into a separate PR, as its scope will grow and it should be thoroughly reviewed by itself.

Reply:
Commit has been moved to #8882